File: C++/A/415A-mashmokhandlights.cpp
#include <bits/stdc++.h>
using namespace std;

int main()
{
    int n, m;
    cin >> n >> m;
    vector<bool> lights(n, true);
    vector<int> whoturnedoff(n, 0), pressed;
    for (int i = 0; i < m; i++)
    {
        int x;
        cin >> x;
        pressed.push_back(x);
    }
    for (const auto &x : pressed)
    {
        for (int i = x - 1; i < n; i++)
        {
            if (lights[i])
            {
                lights[i] = false;
                whoturnedoff[i] = x;
            }
            else
                break;
        }
    }
    for (int elem : whoturnedoff)
        cout << elem << ' ';
}

File: Python/A/415A-mashmokhandlights.py
n, m = map(int, input().split())
lights = [True] * n
whoturnedoff = [0] * n
pressed = list(map(int, input().split()))
for button in pressed:
    for i in range(button - 1, n):
        if lights[i]:
            lights[i] = False
            whoturnedoff[i] = button
        else:
            break
print(*whoturnedoff)
